myadmin-angular/projects/forgot-studentnumber-app/src/app/components/forgot-student-search/forgot-student-search.component.ts

```diff
@@ -0,0 +1,131 @@
+import {ToasterNotificationService, StudentInfo} from "myadmin-lib";
+import {SearchCriteriaService, StudentService} from "../../services";
+
+@Component({
+  selector: 'unisa-forgot-student-search',
+  templateUrl: './forgot-student-search.component.html',
+  styleUrls: ['./forgot-student-search.component.scss']
+})
+export class ForgotStudentSearchComponent implements OnInit {
+
+  studentInputForm:FormGroup;
+
+  studentInfo:StudentInfo;
+
+  /**
+   * Reference to blockUI
+   */
+  @BlockUI()
+  private blockUI:NgBlockUI;
+
+  constructor(private router:Router,
+              private formBuilder:FormBuilder,
+              private searchCriteriaService:SearchCriteriaService,
+              private studentService:StudentService,
+              private datePipe:DatePipe,
+              private toaster:ToasterNotificationService) {
+    this.initForm();
+  }
+
+  private initForm() {
+    this.studentInputForm = this.formBuilder.group({
+      surname: [null, Validators.required],
+      firstNames: [null, Validators.required],
+      dateOfBirth: [null, Validators.required],
+      identityNumber: [null],
+      searchOption: [''],
+      passportNumber: [null]
+    });
+  }
+
+  ngOnInit() {
+  }
+
+  get surname():FormControl {
+    return <FormControl>this.studentInputForm.controls['surname'];
+  }
+
+  get firstNames():FormControl {
+    return <FormControl>this.studentInputForm.controls['firstNames'];
+  }
+
+  get dateOfBirth():FormControl {
+    return <FormControl>this.studentInputForm.controls['dateOfBirth'];
+  }
+
+  get searchOption():FormControl {
+    return <FormControl>this.studentInputForm.controls['searchOption'];
+  }
+
+  get identityNumber():FormControl {
+    return <FormControl>this.studentInputForm.controls['identityNumber'];
+  }
+
+  get passportNumber():FormControl {
+    return <FormControl>this.studentInputForm.controls['passportNumber'];
+  }
+
+
+  private studentCall(students:StudentInfo[]):any {
+    if (students.length > 1) {
+      this.toaster.error('Validation Error', 'There is more than one student number that complies with the information submitted. Please contact study-info@unisa.ac.za for assistance.');
+      this.blockUI.stop();
+      return;
+    }
+    if (students.length == 0) {
+      this.toaster.error('Validation Error', 'Student number not found. Check that your details have been entered correctly. If correct, please contact study-info@unisa.ac.za for assistance.');
+      this.blockUI.stop();
+      return;
+    }
+    this.searchCriteriaService.studentInfo = {...students[0]};
+    this.router.navigateByUrl('/result')
+  }
+
+  onSubmit() {
+    if (this.validateForm()) {
+      this.blockUI.stop();
+      return;
+    }
+    this.studentInfo = {...this.studentInputForm.value};
+
+    const dateString = this.datePipe.transform(this.studentInfo.dateOfBirth, 'yyyy-MM-dd', null, null);
+
+    this.blockUI.start("Loading information...");
+    if (this.identityNumber.value) {
```

Comment:
I see here is where you used the `get` for the field on the form. I guess it is probably fine, but what the `get`ters above does is it also leaks out the value as a `public` field, and also makes it available to the HTML which I don't think is what you intented. You could (just a suggestion) instead add a line(s) like this above in this method

```javascript
const formValue:any = this.studentInputForm.value
```
or if the form directly maps a know type (e.g StudentInfo)
```javascript
const formValue:StudentInto = this.studentInputForm.value
```

and then reference the value that you are interested in e.g. `formValue.identityNumber`